src/Discovery/src/ClientBase/ConfigurationUrlHelpers.cs

```diff
@@ -19,7 +20,7 @@ public static List<Uri> GetAspNetCoreUrls(this IConfiguration config)
         if (!string.IsNullOrEmpty(urls))
         {
             var addresses = urls.Split(';');
-            foreach (var address in addresses)
+            foreach (var address in addresses.Where(a => !string.IsNullOrEmpty(a)))
```

Comment:
Why not just specify the [split behavior](https://learn.microsoft.com/en-us/dotnet/api/system.string.split?view=net-6.0#system-string-split(system-char-system-stringsplitoptions)) directly?

```c#
var addresses = urls.Split(';', StringSplitOptions.RemoveEmptyEntries);
```

It's probably not so relevant here, but these things often get vectorized for maximum performance. Expressing intent instead of implementation enables the runtime to optimize such calls (today or in the future).

There's also a flag to trim entries, though I don't know if we need that here.